engine/src/main/java/io/camunda/zeebe/engine/state/message/PendingMessageSubscriptionState.java

```diff
@@ -36,7 +39,16 @@ public void visitSubscriptionBefore(
               commandEntry.getElementInstanceKey(),
               BufferUtil.wrapString(commandEntry.getMessageName()));
 
-      visitor.visit(subscription);
+      if (subscription == null) {
+        // This case can occur while a scheduled job is running asynchronously
+        // and the stream processor removes one of the returned subscriptions from the state.
+        LOG.warn(
+            "Expected to find a subscription with key {} and message name {}, but none found. The state is inconsistent.",
+            commandEntry.getElementInstanceKey(),
+            commandEntry.getMessageName());
+      } else {
```

Comment:
🔧 you can remove the else state by returning at the end of the if state

Reply:
Below copied from original PR comment:

@nicpuppa you probably mean using `continue` here. Because if I use `return` statement, it will go out of the loop and we will skip other subscriptions. Since, using `continue` is not a good practice of clean code, I prefer `else` statement instead. WDYT?

Reply:
Didn't saw it was a loop 👀 👓 , sorry for that!

In this particular case using continue statement is not so bad, but is just my own preference. You can ignore my suggestion